mail/backup_mailer/src/opnsense/mvc/app/library/OPNsense/Backup/Mailer.php

```diff
@@ -0,0 +1,228 @@
+            $mailer->serializeToConfig();
+            Config::getInstance()->save();
+        }
+        return $validation_messages;
+    }
+
+    /**
+     * @return array filelist
+     */
+    public function backup()
+    {
+        $cnf = Config::getInstance();
+        $mailer = new MailerSettings();
+        if ($cnf->isValid() && !empty((string)$mailer->MailEnabled)) {
+            $confdata = file_get_contents('/conf/config.xml');
+            $result = self::sendEmail($mailer, $confdata);
+        }
+
+        return array($result);
+    }
+
+    /**
+     * Is this provider enabled
+     * @return boolean enabled status
+     */
+    public function isEnabled()
+    {
+        $mailer = new MailerSettings();
+        return (string)$mailer->MailEnabled === "1";
+    }
+
+    public function sendEmail($config, $confdata)
+    {
+        $smtpUsername = (string)$config->SmtpUsername;
+        $smtpPassword = (string)$config->SmtpPassword;
+        $gpgPublicKey = (string)$config->GpgPublicKey;
+        $gpgEmail     = (string)$config->GpgEmail;
+
+        $date     = date('Y-m-d');
+        $hostname = gethostname();
+
+        PHPMailerAutoload(PHPMailer);
+        $mail = new \PHPMailer(true);
+        $mail->IsHTML(true);
+        $mail->IsSMTP();
+        $mail->SetFrom($gpgEmail);
+        $mail->AddAddress((string)$config->Receiver);
+        $mail->Host    = (string)$config->SmtpHost;
+        $mail->Port    = (string)$config->SmtpPort;
+        $mail->Subject = $hostname . ' OPNsense config backup ' . $date;
+        $mail->Body    = $hostname . ' config backup file';
+
+        if ((string)$config->SmtpSSL === "1") {
+            $mail->SMTPSecure = 'ssl';
+        } else {
+            $mail->SMTPOptions = array(
+                'ssl' => array(
+                    'verify_peer' => false,
+                    'verify_peer_name' => false,
+                    'allow_self_signed' => true,
+                ),
+            );
+        }
+
+        if ($smtpUsername != "" && $smtpPassword != "") {
+            $mail->SMTPAuth = true;
+            $mail->Username = $smtpUsername;
+            $mail->Password = $smtpPassword;
+        }
+
+        self::import_key($gpgPublicKey);
+
+        self::encrypt_data($confdata, $gpgEmail);
+
+        $attachmentName = 'config_' . gethostname() . '_' . $date . '.xml.asc';
+        $mail->AddAttachment('backup', $attachmentName);
+
+        $mail->Send();
+
+        return $attachmentName;
+    }
+
+    function import_key($gpgPublicKey) {
+        $gpgPublicKeyFile = "key.asc";
+
+        if ($gpgPublicKey != "") {
+            file_put_contents($gpgPublicKeyFile, $gpgPublicKey);
+
+            $descriptorspec = array(
+                0 => array("pipe", "r"),
+                1 => array("pipe", "w"),
+                2 => array("file", "/tmp/import_key_error.txt", "a")
+            );
+
+            $process = proc_open('gpg --import < ' . $gpgPublicKeyFile, $descriptorspec, $pipes);
```

Comment:
remove the shell pipe - just execute `gpg --import` and write the public key file to `$pipes[1]`